### PR TITLE
feat: add BTC to on-ramp & off-ramp assets

### PR DESCRIPTION
**Pull request by josheleonard**
Resolves https://github.com/brave/brave-browser/issues/35204

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See STR in issue

### CODE REVIEW

File: components/brave_wallet/browser/asset_ratio_service.cc [hunk 1]
```diff
@@ -19,6 +19,7 @@
 #include "brave/components/brave_wallet/browser/asset_ratio_response_parser.h"
 #include "brave/components/brave_wallet/browser/brave_wallet_constants.h"
 #include "brave/components/brave_wallet/browser/json_rpc_requests_helper.h"
+#include "brave/components/brave_wallet/common/brave_wallet.mojom-forward.h"
```
Comment (supermassive) [inline components/brave_wallet/browser/asset_ratio_service.cc:22]: ~#include "brave/components/brave_wallet/common/brave_wallet.mojom-forward.h"~

**Review by Douglashdaniel — APPROVED** [review]
++

File: components/brave_wallet_ui/page/screens/fund-wallet/fund-wallet.tsx [hunk 2]
```diff
@@ -682,6 +690,34 @@ function PurchaseOptionSelection({ isAndroid }: Props) {
...
+        }
+      }
+    }
+
+    // cleanup
+    return () => {
+      ignore = true
+    }
+  }, [selectedAccount?.accountId])
+
```
Comment (supermassive) [inline components/brave_wallet_ui/page/screens/fund-wallet/fund-wallet.tsx:720]: Maybe recently added `useReceiveAddressQuery` hook instead? :)

**Review by supermassive — APPROVED** [review]

**Review by supermassive — APPROVED** [review]

**Review by supermassive — APPROVED** [review]

**Review by supermassive — APPROVED** [review]

**Comment by srirambv** [comment]
Verification passed on

Brave | 1.64.26 Chromium: 121.0.6167.85 (Official Build) nightly (64-bit)
-- | --
Revision | 3cd62d605d5c48e98d2820fe4aee84aa35fb525f
OS | Linux

- Verified steps from issue description
- Verified BTC is available as an option in on-ramp and off-ramp 

On-Ramp | Off-Ramp
-- | --
<video src=https://github.com/brave/brave-core/assets/17010094/9b352726-4644-4471-b4ed-ff349015c0ff> | <video src=https://github.com/brave/brave-core/assets/17010094/601fe269-636f-4cb0-873b-c7e55cab16ba>

**Pull request merged**